### PR TITLE
BLD: Added the pyenvbuilder launcher - a console_scripts entry at setup.py

### LINKED ISSUE #8: pyenvbuilder launcher
Have a `console_scripts` entry at setup.py so we can have `pyenvbuilder` as a script at the shell.

### PR DESCRIPTION
**Pull request by cristinasewell**
Added a **console_scripts** entry point for `pyenvbuilder`. We should be able to install and run `pyenvbuilder` as a script at the shell, as well as checking the version  or for help:
```
$ pyenvbuilder
$ pyenvbuilder --version # or pyenvbuilder --V 
$ pyenvbuilder -h
```

Closes #8

### CODE REVIEW

File: pyenvbuilder/commands.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import pyenvbuilder
+
+def add_commands():
+    '''
+    Defines shell commands
+    '''
+
+    parser = argparse.ArgumentParser(description='Python Environment Builder')
+
+    parser.add_argument('--V', '--version', action='version',
```
Comment (cristinasewell) [inline pyenvbuilder/commands.py]: I just realized this, but I think i should change the option `--V` to just `-V` - single dash

File: pyenvbuilder/commands.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+'''
+pyenvbuilder's commands
+'''
+
+import argparse
+import sys
+import pyenvbuilder
+
+def add_commands():
```
Comment (hhslepicka) [inline pyenvbuilder/commands.py]: PEP8 - E302 expected 2 blank lines, found 1

File: pyenvbuilder/commands.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    '''
+    Defines shell commands
+    '''
+
+    parser = argparse.ArgumentParser(description='Python Environment Builder')
+
+    parser.add_argument('--V', '--version', action='version',
+                        version='pyenvbuilder {version}'.format(version=pyenvbuilder.__version__),
```
Comment (hhslepicka) [inline pyenvbuilder/commands.py]: PEP8 - E501 line too long (98 > 79 characters)

File: pyenvbuilder/commands.py (hunk 1 above)
Comment (hhslepicka) [reply pyenvbuilder/commands.py]: Why not just --version?

File: pyenvbuilder/commands.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                        help='show version and exit')
+
+    parser.parse_args()
+
```
Comment (hhslepicka) [inline pyenvbuilder/commands.py]: PEP8 - W293 blank line contains whitespace

File: pyenvbuilder/main.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+'''
+penvbuilder entry point
+'''
+from pyenvbuilder.commands import add_commands
+
+def main():
```
Comment (hhslepicka) [inline pyenvbuilder/main.py]: PEP8 - E302 expected 2 blank lines, found 1

**Review by hhslepicka — CHANGES_REQUESTED** [review]
Great job adding the console_script.

Here are a couple of suggestions:

- [x] Move the `add_commands` into `main.py` and nuke commands.py.
    The argparse parts will only be useful for the `console_script` so it belongs into main.

- [x] Maybe rename `add_commands` to `parse_arguments` and return the result of the parser?

- [x] Make it possible to run via code with a pattern like this at `main.py`:
```python
    def launch(arg1, arg2, arg3):
        ...

    def main():
        args = parse_arguments()
        kwargs = vars(args)
        launch(**kwargs)
```
    This will allow users to import and run the central code but specifying the options via code.

- [x] Add a `__main__.py` file which will import `main.py` and call `main()`
    This will allow users to do something like this:
    `python -m pyenvbuilder ...`
    And it will have the same effect as the console script. `pip` offers the same possibility.

**Comment by hhslepicka** [comment]
For the PEP8 issues, you can install `Flake8` https://flake8.pycqa.org/en/latest/ like this:
`pip install flake8`
And running as so:
```
flake8 pyenvbuilder
```
In which `pyenvbuilder` is your code folder.

File: pyenvbuilder/commands.py (hunk 1 above)
Comment (cristinasewell) [reply pyenvbuilder/commands.py]: Yeah that sounds good :)

File: pyenvbuilder/main.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+'''
+import argparse
+import logging
+from pyenvbuilder import __version__
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def launch(log_level="INFO"):
```
Comment (cristinasewell) [inline pyenvbuilder/main.py:12]: For now I just have the `log_level` as an argument but I will be adding more as I develop this.

File: pyenvbuilder/main.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    '''
+    PyEnvBuilder's Launcher
+    '''
+    pyenvbuilder_logger = logging.getLogger('')
```
Comment (hhslepicka) [inline pyenvbuilder/main.py]: Recently I learned that we should never attach and switch levels at the root logger ('').
Would you please change it to:
```
    pyenvbuilder_logger = logging.getLogger('pyenvbuilder')
```

This way when we change the level and also attach the handler we will do so just for our package and not for other dependent packages that we have (which would be affected if we use `.getLogger('')`.

File: pyenvbuilder/main.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
+'''
+pyenvbuilder's commands
+'''
+import argparse
+import logging
+from pyenvbuilder import __version__
+
+
+LOGGER = logging.getLogger(__name__)
```
Comment (hhslepicka) [inline pyenvbuilder/main.py]: We should go with lowercase `logger` as it is not a constant.

**Review by hhslepicka — CHANGES_REQUESTED** [review]
All looks good to me module the two little change requests. Thank you for taking care of it.

File: pyenvbuilder/main.py (hunk 7 above)
Comment (cristinasewell) [reply pyenvbuilder/main.py]: Ok great, thank you for letting me know, I implemented it

File: pyenvbuilder/main.py (hunk 8 above)
Comment (cristinasewell) [reply pyenvbuilder/main.py]: got it

**Review by hhslepicka — APPROVED** [review]

**Pull request merged**